tests/unit/test_link_generation_facade.py

```diff
@@ -0,0 +1,130 @@
+
+TEST_FILE = 'tests/totally_real/verify_I_am_good_link.py'
+
+
+# ======================================================================================================================
+# Test Suites
+# ======================================================================================================================
+class TestLinkGenerationFacade(object):
+    """Tests for the LinkGenerationFacade"""
+
+    def test_asc_failure_link(self, mocker):
+        """Validate when configured with asc as ci-environment"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = FAILURE_OUTPUT
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py#L23'  # noqa
+
+    def test_asc_def_link(self, mocker):
+        """Validate fallback to def line number"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = 'I dont contain an assert message'
+        zztl.def_line_number = '88'
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py#L88'  # noqa
+
+    def test_asc_file_link(self, mocker):
+        """Validate fallback to file with no line number"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = 'I dont contain an assert message'
+        zztl.def_line_number = ''
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py'  # noqa
+
+    def test_mk8s(self, mocker):
+        """Validate when configured with mk8s as ci-environment"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'mk8s'
+        zz.testsuite_props = MK8S_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = FAILURE_OUTPUT
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/mk8s/blob/master/tools/installer/tests/totally_real/verify_I_am_good_link.py#L23'  # noqa
```

Comment:
See comment for line 58.

I do not know where the `tools/installer` path came from.

Reply:
I'll comment in the code for this one.  
Its specific to the mk8s repo and can not be derived from the XML at this time.